pyproject.toml

```diff
@@ -2,34 +2,37 @@
 name = "stapi-fastapi"
 # placeholder version filled by poetry-dynamic-versioning
 version = "0.0.0"
-description = "Spatio Temporal Asset Tasking with FastAPI"
+description = "Sensor Tasking API (STAPI) with FastAPI"
 authors = ["Christian Wygoda <christian.wygoda@wygoda.net>"]
 license = "MIT"
 readme = "README.md"
 packages = [{include = "stapi_fastapi", from="src"}]
 
 [tool.poetry.dependencies]
-python = "^3.12.0"
-fastapi = "^0.115.0"
-pydantic = "^2.10.1"
-geojson-pydantic = "^1.1.1"
-pygeofilter = "^0.2.4"
-returns = "^0.23.0"
+python = ">=3.12,<4.0"
+fastapi = ">=0.115.0"
```

Comment:
shouldn't these all include a "but less than next major" condition as well?

Reply:
I had to add it for python because some other package had that constraint, and it wouldn't build without also applying it to python. 

The argument for only having a floor is that it's up to the downstream user to validate that it works with whatever specific fastapi/etc version they choose for their app, and that we don't want to unnecessarily restrict the version they can use just because we're not sure it will work. We're also not testing this but with one version of any of these libraries, so it's still not guaranteed that it will work with whatever version the downstream chooses even if it's within some reasonable semantic versioning constraints.

Reply:
It would be nice to have CI run tests on an install with all dependencies pinned to their minimum version and on an install with latest set of dependency versions that resolve. But I think that is out of scope for this ticket.